lib/monocle.js

```diff
@@ -118,7 +118,7 @@ var _handle = function(method, path, options) {
             // Check cache if attempting to get resource
             var cached = this._cache.get(path);
             if (cached) {
-                if (!options || !options.props) {
+                if (!options || !options.props || !options.query) {
```

Comment:
If you pass in `options.props` but no `options.query`, this condition will be satisfied and the symlink will be resolved without props. Logic should be:

    (!options || (!options.props && !options.query))